Sources/ConcordiumSwiftSdk/NodeClient.swift

```diff
@@ -3,11 +3,23 @@ import GRPC
 import NIOCore
 import NIOPosix
 
-public class Client {
-    let grpc: Concordium_V2_QueriesNIOClient
+public protocol NodeClient {
+    func getCryptographicParameters(at block: BlockIdentifier) async throws -> CryptographicParameters
 
-    public init(channel: GRPCChannel) {
-        grpc = Concordium_V2_QueriesNIOClient(channel: channel)
+    func getNextAccountSequenceNumber(of address: AccountAddress) async throws -> NextAccountSequenceNumber
+
+    func getAccountInfo(of account: AccountIdentifier, at block: BlockIdentifier) async throws -> AccountInfo
+}
+
+public class GrpcNodeClient: NodeClient {
```

Comment:
While this class is _internally_ wrapping grpc, I don't think we should expose that fact via the type name, since we are trying our best to hide away all grpc details from the users of this class.
Yes, if the intention is to later add additional clients to support other protocols - and this client would just be one of the options - then the name could make sense, but if it is intended as _the_ client, maybe it would be better if we called it something like `Default/Standard/Real/etc... ConcordiumClient`

Reply:
*Consuming* code will use the `ConcordiumNodeClient` protocol which hides the underlying implementation. The concrete type is only seen by *constructing* code, and I don't want the fact that it's based on gRPC to be hidden there. So I'd like to keep it as it is :)